src/main/java/uk/ac/ic/wlgitbridge/bridge/db/postgres/PostgresDBStore.java

```diff
@@ -0,0 +1,279 @@
+    } catch (Exception e) {
+      Log.error("Error connecting to Postgres: {}", e.getMessage());
+      throw new DBInitException(e);
+    }
+  }
+
+  @Override
+  public int getNumProjects() {
+    try (
+      Connection connection = pool.getConnection();
+      Statement statement = connection.createStatement();
+         ResultSet rs = statement.executeQuery("SELECT count(*) from projects;\n")) {
+      if (rs.next()) {
+        int result = rs.getInt(1);
+        return result;
+      } else {
+        return 0;
+      }
+    } catch (Exception e) {
+      Log.error("Error while querying Postgres: {}", e.getMessage());
+      throw new RuntimeException(e);
+    }
+  };
+
+  @Override
+  public List<String> getProjectNames() {
+    try (
+      Connection connection = pool.getConnection();
+      Statement statement = connection.createStatement();
+         ResultSet rs = statement.executeQuery("SELECT name from projects;\n")) {
+      List<String> result = new ArrayList<String>();
+      while (rs.next()) {
+        result.add(rs.getString(1));
+      }
+      return result;
+    } catch (Exception e) {
+      Log.error("Error while querying Postgres: {}", e.getMessage());
+      throw new RuntimeException(e);
+    }
+  };
+
+  @Override
+  public void setLatestVersionForProject(String project, int versionID) {
+    try (
+      Connection connection = pool.getConnection();
+      PreparedStatement statement = connection.prepareStatement(
+        "INSERT INTO "
+           + "projects (name, version_id, last_accessed) "
+           + "VALUES (?, ?, now()) "
+           + "ON CONFLICT (name) DO UPDATE "
+           + "SET version_id = EXCLUDED.version_id, last_accessed = now();\n")) {
+      statement.setString(1, project);
+      statement.setInt(2, versionID);
+      statement.executeUpdate();
+    } catch (Exception e) {
+      Log.error("Error while querying Postgres: {}", e.getMessage());
+      throw new RuntimeException(e);
+    }
+  };
+
+  @Override
+  public int getLatestVersionForProject(String project) {
+    try (
+      Connection connection = pool.getConnection();
+      PreparedStatement statement = connection.prepareStatement(
+        "SELECT version_id from projects where name = ?;")) {
+      statement.setString(1, project);
+      try (ResultSet rs = statement.executeQuery()) {
+        if (rs.next()) {
+          int versionId = rs.getInt(1);
+          return versionId;
+        } else {
+          return 0;
+        }
+      }
+    } catch (Exception e) {
+      Log.error("Error while querying Postgres: {}", e.getMessage());
+      throw new RuntimeException(e);
+    }
+  };
+
+  @Override
+  public void addURLIndexForProject(String projectName, String url, String path) {
+    try (
+      Connection connection = pool.getConnection();
+      PreparedStatement statement = connection.prepareStatement(
+         // TODO: check this is actually correct, with the two indexes
```

Comment:
We've done as much checking as we're going to at this point, here... So probably should remove the `TODO`